compilers/concrete-optimizer/concrete-optimizer/src/optimization/dag/multi_parameters/generic_generation.rs

```diff
@@ -0,0 +1,169 @@
+            let weighted_b = dag.add_linear_noise(
+                [lut_b],
+                LevelledComplexity::ZERO,
+                [def_b.norm2.sqrt()],
+                Shape::number(),
+                "",
+            );
+
+            dag.add_composition(weighted_b, inp_a);
+
+            if generate_fks && def_a > def_b {
+                let inp_a = dag.add_input(def_a.precision, Shape::number());
+                let lut_a = dag.add_lut(inp_a, FunctionTable::UNKWOWN, def_a.precision);
+                let _weighted_a = dag.add_linear_noise(
+                    [lut_a],
+                    LevelledComplexity::ZERO,
+                    [def_a.norm2.sqrt()],
+                    Shape::number(),
+                    "",
+                );
+
+                let inp_b = dag.add_input(def_b.precision, Shape::number());
+                let lut_b = dag.add_lut(inp_b, FunctionTable::UNKWOWN, def_b.precision);
+                let _weighted_b = dag.add_linear_noise(
+                    [lut_b],
+                    LevelledComplexity::ZERO,
+                    [def_b.norm2.sqrt()],
+                    Shape::number(),
+                    "",
+                );
+
+                let _ = dag.add_linear_noise(
+                    [lut_a, lut_b],
+                    LevelledComplexity::ZERO,
+                    [0., 0.],
+                    Shape::number(),
+                    "",
+                );
+            }
+        }
+    }
+
+    let precisions: Vec<_> = partitions.iter().map(|def| def.precision).collect();
+    let n_partitions = precisions.len();
+    let p_cut = PartitionCut::maximal_partitionning(&dag);
+    let config = Config {
+        security_level: 128,
+        maximum_acceptable_error_probability: _4_SIGMA,
+        key_sharing: true,
+        ciphertext_modulus_log: 64,
+        fft_precision: 53,
+        complexity_model: &CpuComplexity::default(),
+    };
+    let search_space = SearchSpace::default_cpu();
+    let cache = decomposition::cache(128, ProcessingUnit::Cpu, None, true, 64, 53);
+    let parameters = optimize(
+        &dag,
+        config,
+        &search_space,
+        &NoSearchSpaceRestriction,
+        &cache,
+        &Some(p_cut),
+        PartitionIndex(0),
+    )
+    .map_or(None, |v| Some(v.1))
+    .unwrap();
+
+    for i in 0..n_partitions {
+        for j in 0..n_partitions {
+            assert!(
+                parameters.micro_params.ks[i][j].is_some(),
+                "Ksk[{i},{j}] missing."
+            );
+            if i > j {
+                assert!(
+                    parameters.micro_params.fks[i][j].is_some(),
+                    "Fksk[{i},{j}] missing."
+                );
+            }
+        }
+    }
+    ExpandedCircuitKeys::of(&parameters).compacted()
+}
+
+#[cfg(test)]
+mod test {
+    use super::{generate_generic_parameters, PartitionDefinition};
+
+    #[test]
+    fn test_generate_generic_parameters() {
+        let _ = generate_generic_parameters(
```

Comment:
Not sure what we test here? Should we check the returned solution and test with other precision partitions?

Reply:
Ah yes, it's a leftover I used during development. I'll add a few checks yes.

Reply:
Actually, it does perform verification since those a re integrated in the generation (all ksk generated + all fks generated).